src/main/java/com/happeningnow/service/factory/ServiceFactory.java

```diff
@@ -0,0 +1,36 @@
+package com.happeningnow.service.factory;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public abstract class ServiceFactory<T, R extends JpaRepository<T, UUID>>{
```

Comment:
it is not a factory. you may name it just as ServiceAbsctract

Reply:
Ok